### PR TITLE
test: lang aladino env unit tests

### PR DESCRIPTION
**Pull request by shay2025**
## Description

This pull request introduces unit tests for `lang/aladino/env`.

## Related issue

*None*

## Type of change

Improvements (non-breaking change without functionality)

## How was this tested?

Ran `task test` cmd.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have ran `task check -f` and have no issues

### CODE REVIEW

**Review by ferreiratiago — CHANGES_REQUESTED** [review]
Please review the use of `reflect.DeepEqual` to make sure it's the right function to compare the output.

File: lang/aladino/env_test.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+		"emptyFunction": aladino.BuildFunctionType([]aladino.Type{}, nil),
+		"emptyAction":   aladino.BuildFunctionType([]aladino.Type{}, nil),
+	})
+
+	gotTypeEnv := *aladino.NewTypeEnv(mockedEnv)
+
+	assert.Equal(t, wantTypeEnv, gotTypeEnv)
+}
+
+// The MockDefaultEnv function to create a mocked default env, it uses the NewEvalEnv which is the function we wish to test
```
Comment (marcelosousa) [inline lang/aladino/env_test.go:174]: This comment is a bit confusing. From the name of the function, we should be testing NewEval but it's only called indirectly through the MockDefaultEnv. It seems that what we are really testing is the MockDefaultEnv.

File: lang/aladino/env_test.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+				}),
+			),
+		},
+		nil,
+	)
+
+	assert.Equal(t, err.(*github.ErrorResponse).Message, failMessage)
+}
+
+func TestNewEvalEnv_WhenNewFileFails(t *testing.T) {
```
Comment (marcelosousa) [inline lang/aladino/env_test.go:196]: Same comment from previous function applies here and in the next test functions.

**Review by marcelosousa — CHANGES_REQUESTED** [review]
Thanks for this! I've left comments in the diff.

**Review by marcelosousa — COMMENTED** [review]
Please check the comments.

File: lang/aladino/mocks.go [hunk 3]
```diff
@@ -15,19 +15,23 @@ import (
...
-const defaultMockPrID = 1234
-const defaultMockPrNum = 6
-const defaultMockPrOwner = "foobar"
-const defaultMockPrRepoName = "default-mock-repo"
+const DefaultMockPrID = 1234
+const DefaultMockPrNum = 6
+const DefaultMockPrOwner = "foobar"
+const DefaultMockPrRepoName = "default-mock-repo"
+
+var DefaultContext = context.Background()
```
Comment (marcelosousa) [inline lang/aladino/mocks.go:27]: please rename to DefaultMockX

**Review by marcelosousa — APPROVED** [review]
🙌

**Pull request merged**